pkg/apis/core/validation/validation.go

```diff
@@ -5627,12 +5615,12 @@ func ValidateWhenUnsatisfiable(fldPath *field.Path, action core.UnsatisfiableCon
 	return nil
 }
 
-// ValidateSpreadConstraintPair tests that if `pair` exists in `existingConstraintPairs`.
-func ValidateSpreadConstraintPair(fldPath *field.Path, pair spreadConstraintPair, existingConstraintPairs []spreadConstraintPair) *field.Error {
-	for _, existingPair := range existingConstraintPairs {
-		if pair.topologyKey == existingPair.topologyKey &&
-			pair.whenUnsatisfiable == existingPair.whenUnsatisfiable {
-			return field.Duplicate(fldPath, pair)
+// ValidateSpreadConstraintNotRepeat tests that if `pair` exists in `existingConstraintPairs`.
+func ValidateSpreadConstraintNotRepeat(fldPath *field.Path, constraint core.TopologySpreadConstraint, restingConstraints []core.TopologySpreadConstraint) *field.Error {
```

Comment:
Why pass fldPath instead of putting it statically in this function.

Reply:
I'd say that's a convention to encapsulate an error in a stateless manner.